### ISSUE TITLE
VSTS IsHostedAgent detection is flawed

### ISSUE THREAD

**Issue by trailmax** [cake-build/cake#1684]
When running CakeBuild in VSTS and rely on `Context.TFBuild().IsRunningOnVSTS`. However we have 2 hosted build agents and second build agent is called "Hosted Agent 2". And build is always failing on the second agent. 

[Here](https://github.com/cake-build/cake/blob/d8f7fe179bdf41fa9d2dbc426a3bd6fbcef7dcfc/src/Cake.Common/Build/TFBuild/TFBuildProvider.cs#L82) code insists that agent name is exactly "Hosted Agent". 

Also We are setting up private build agent and [this code](https://github.com/cake-build/cake/blob/d8f7fe179bdf41fa9d2dbc426a3bd6fbcef7dcfc/src/Cake.Common/Build/TFBuild/TFBuildProvider.cs#L45) does not agree with private build agents. It is still VSTS build, only not run in hosted agent. 

I propose to change the check for the agent name to `.StartsWith("Hosted Agent")` and remove check for `IsHostedAgent` from `IsRunningOnVSTS` option.

@agc93 what do you think? (as far as I remember you have done work on this)

I can do PR for this.

**Comment by trailmax** [comment]
Looking on my VSTS, my other hosted agent is called "Hosted VS2017 2". So we can't rely on this value to be anything consistent.

**Comment by patriksvensson** [comment]
Related to #1580.

**Comment by agc93** [comment]
Yep, I'll have a look at this (a few different parts of the logic need improvements)

**Comment by volak** [comment]
Just ran into this issue - only need to know if its building on VSTS so now I'm doing 
`var isVSTS = buildSystem.TFBuild.IsRunningOnVSTS || buildSystem.TFBuild.IsRunningOnTFS;`
instead of just `IsRunningOnVSTS `

**Comment by kevbite** [comment]
We've just stumbled across this issue too since we've added a couple more hosted agents to our team. As a workaround we're just checking that we've got a hosted name:
```
var isRunningOnVSTS = !string.IsNullOrEmpty(EnvironmentVariable("AGENT_NAME"));
```

**Comment by augustoproiete** [comment]
@agc93 / @pascalberger Has this been resolved by [`8d3483b`](https://github.com/pascalberger/cake/commit/8d3483bc9c507f33f4bdbffca9265f989914b900)?

**Comment by devlead** [comment]
@augustoproiete yes, going ahead and closing this issue.
